### DISCUSSION TITLE
Would the community recommend using Cake for ARM template builds?

### DISCUSSION THREAD

**Discussion by DougalKennedy** [cake-build/cake#3085] [Q&A]
Currently I have a solution with some code producing ARM template JSON. I have an Azure devOps pipeline to execute the code, building the templates, validating the templates and finally deploying the actual infrastructure and other Azure resources.

Having been introduced to Cake by @gep13 last night (great presentation btw) the appeal to use it is obvious, I'm just wondering how close to the InfraAsCode world I can get here?

I guess as  a minimum I could be using the [Azure Resource Manager Template Toolkit](https://github.com/Azure/arm-ttk) as a build task in the Cake script?

**Comment by gitfool** [comment]
You might want to have a look at [Pulumi](https://www.pulumi.com/azure/) first, which has dotnet/c# support, then try to drive it with [Cake.Pulumi](https://cakebuild.net/dsl/pulumi/).
